Tests/Regex/DPLRegularExpressionSpec.m

```diff
@@ -33,7 +33,31 @@
                                                         @"this": @"thisvalue",
                                                         @"that": @"thatvalue" });
     });
-    
+
+    it(@"should match an array", ^{
+        DPLRegularExpression *expression = [DPLRegularExpression regularExpressionWithPattern:@"/hello/:list"];
+
+        DPLMatchResult *matchResult = [expression matchResultForString:@"/hello/list[]=item1&list[]=item2"];
+        expect(matchResult.match).to.beTruthy();
+        expect(matchResult.namedProperties).to.equal(@{ @"list": @[ @"item1", @"item2" ] });
```

Comment:
We should not parse path components as a query. Based on the lack of query indicator `?` I would expect `@{ @"list": @"list[]=item1&list[]=item2" }`